src/config/ch.sbb.direktverbindungen/DvFeatureInfo/DvFeatureInfo.js

```diff
@@ -0,0 +1,271 @@
+        left: 0,
+        pointerEvents: 'none',
+        backgroundImage:
+          'linear-gradient(to bottom, rgba(255,255,255,0), rgba(255,255,255, 1) 90%)',
+        width: '100%',
+        height: '10em',
+      },
+      '& .open': {
+        '& .wkp-menu-item-header-toggler': {
+          transform: 'rotate(180deg)',
+        },
+      },
+    },
+    featureInfos: {
+      maxHeight: '100%',
+      overflow: 'auto',
+    },
+    featureInfoItem: {
+      marginLeft: 38,
+      marginRight: 16,
+    },
+    imageLine: {
+      '& img': {
+        width: '100%',
+      },
+    },
+  };
+});
+
+function DvFeatureInfo({ filterByType }) {
+  const featureInfo = useSelector((state) => state.app.featureInfo);
+  const layers = useSelector((state) => state.map.layers);
+  const [infoKey, setInfoKey] = useState();
+  const [teaser, setTeaser] = useState(true);
+  const [expandedHeight, setExpandedHeight] = useState();
+  const classes = useStyles({ teaser });
+  const teaserOnClick = useCallback(() => {
+    return teaser ? setTeaser(false) : undefined;
+  }, [teaser]);
+  const [revision, forceRender] = useState();
+
+  const getVisibleLayerKeys = useCallback(
+    () =>
+      layers
+        .filter((l) => DV_DAY_NIGHT_REGEX.test(l.key) && l.visible)
+        .map((l) => l.key),
+    [layers],
+  );
+
+  const dvMainLayer = useMemo(
+    () => layers.find((l) => l.key === `${DV_KEY}.main`),
+    [layers],
+  );
+  const [layersVisible, setLayersVisible] = useState(getVisibleLayerKeys());
+
+  const dvFeatures = useMemo(() => {
+    const features = featureInfo.reduce((feats, info) => {
+      info.features.forEach((feat) => feat.set('layer', info.layer));
+      return [...feats, ...info.features];
+    }, []);
+    features.sort((feat) => (feat.get('line') === 'night' ? -1 : 1));
+    const cleaned = removeDuplicates(parseDvFeatures(features)).filter(
+      (feat) => !!feat.get('mapboxFeature'),
+    );
+    return filterByType
+      ? cleaned.filter(
+          (feat) =>
+            !!layersVisible.find((layerKey) => {
+              return `${DV_KEY}.${feat.get('line')}` === layerKey;
+            }),
+        )
+      : cleaned;
+  }, [featureInfo, layersVisible, filterByType]);
+
+  const previousFeatureInfo = usePrevious(featureInfo);
+  const previousDvFeatures = usePrevious(dvFeatures);
+  const previousInfoKey = usePrevious(infoKey);
+
+  useEffect(() => {
+    if (featureInfo !== previousFeatureInfo) {
+      setTeaser(true);
+      setInfoKey(getId(dvFeatures[0]));
+      return;
+    }
+    if (dvFeatures !== previousDvFeatures) {
+      const previousSelectedFeature = dvFeatures.find(
+        (feat) => getId(feat) === previousInfoKey,
+      );
+      if (previousInfoKey) {
+        setInfoKey(previousSelectedFeature && getId(previousSelectedFeature));
+      }
+    }
+  }, [
+    dvFeatures,
+    previousDvFeatures,
+    previousInfoKey,
+    layersVisible,
+    featureInfo,
+    previousFeatureInfo,
+  ]);
```

Comment:
Split this useEffect in 2 useEffect it makes it more clear , and you don't have to care about comparing to previous stuff.

Reply:
This code handles various cases:
- When you change the feature info via map click (or the list button) it should always select the first line and set teaser to true
- When you have selected a line but the filter it out via the layer switcher NO line should be selected (e.g. you select a night line, then disable the night layer and only day layers remain)
- If the list is updated via the layer switchers and the selected line is still in the list, that one should also stay selected
I think these cases is difficult to handle in separate useEffects and without the usePrevious hook. We can discuss further if you want...